frontend/packages/ceph-storage-plugin/src/components/modals/add-capacity-modal/add-capacity-modal.tsx

```diff
@@ -34,18 +35,30 @@ import {
   storageClassTooltip,
   defaultRequestSize,
 } from '../../../constants';
-import { OCSStorageClassDropdown } from '../storage-class-dropdown';
+import { filterSC, isArbiterSC, isValidTopology } from '../../../utils/install';
 import { PVsAvailableCapacity } from '../../ocs-install/pvs-available-capacity';
 import { pvResource, nodeResource } from '../../../resources';
 import { createDeviceSet, getDeviceSetCount } from '../../ocs-install/ocs-request-data';
 import { DeviceSet } from '../../../types';
 import './add-capacity-modal.scss';
-import { isArbiterSC, isValidTopology } from '../../../utils/install';
+
 import { checkArbiterCluster, checkFlexibleScaling } from '../../../utils/common';
 
 const queries = (() => Object.values(CAPACITY_INFO_QUERIES))();
 const parser = compose((val) => val?.[0]?.y, getInstantVectorStats);
 const getProvisionedCapacity = (value: number) => (value % 1 ? (value * 3).toFixed(2) : value * 3);
+/**
+ * By default, returns the storageClass used for initial installation.
+ * In case of multiple deviceSets, returns the last deviceSets's storageClass.
+ * Otherwise, returns the name of the storageClass selected from the dropdown.
+ */
+const getSelectedKey = (
+  storageClass: StorageClassResourceKind,
+  deviceSets: DeviceSet[],
+): string => {
+  if (storageClass) return getName(storageClass);
+  return deviceSets?.[deviceSets.length - 1]?.dataPVCTemplate?.spec?.storageClassName;
```

Comment:
There is already - `selectedSCName` we are calculating above, lets reuse that.
```suggestion
  return selectedSCName || deviceSets?.[0]?.dataPVCTemplate?.spec?.storageClassName;
```